rfc/rfc-92/rfc-92.md

```diff
@@ -0,0 +1,102 @@
+
+# RFC-92: Pluggable Table Formats in Hudi
+
+## Proposers
+
+*   Balaji Varadarajan
+
+## Approvers
+
+*   Vinoth Chandar
+*   Ethan Guo
+
+## Status
+
+JIRA: <TBD>
+
+## Abstract
+
+This RFC proposes support for different backing table format implementations inside Hudi. For the past 4 years at-least, we have been consistently defining Hudi as a broader platform and software [stack](https://hudi.apache.org/docs/hudi_stack) that delivers much of these benefits. Hudi's table format makes choices specific to data lake workloads, allowing efficient read/write (even the recent [blog](https://bytearray.substack.com/p/computer-science-behind-lakehouse) from Vinoth), has major differences and advantages compared to other approaches. The community plans to centrally focus on the native Hudi storage format.
+
+However, there may be benefits to allowing other storage layouts/table formats to fit under Hudi's higher level functionality. This also has non-technical benefits of insulating the project from vendor marketing wars. Most contributors (such as myself) are happily part of the global Hudi open-source community, for the sake of just building technology.
+
+## Background
+
+Expanding further, there are plenty of valid technical reasons on why Hudi should allow different storage layouts, under the upper layer reader/writer and table services implementations.
+
+1. We have use-cases, for cloud-native/high performance implementations of timeline (\`HoodieTimeline\`) and metadata (\`HoodieMetadata\` interface). In our use-case, we would like to explore backing them using NoSQL datastore like DynamoDB, for ultra-low latency queries.
```

Comment:
this is a valid technical reason. Something I was/am hoping the metaserver/timeline server standalone solves. But seems like you want to just not run another server.. and instead plug in cloud native services?

Reply:
yes.  metaserver can solve it. but I feel its easier to just have pluggable interfaces to talk to already readily available cloud services. Reduces ops cost/burden of running one more service. I got this idea when I was making the timeline abstractions for v1 and v2.

Reply:
The community would like to work on the metaserver in the long term.  Maybe some of the interfaces are useful in the metaserver perspective.  For example, the metaserver can leverage the DynamoDB-based implementation for the timeline.